### ISSUE TITLE
Can python decorator automatically/locally turn on/off rule?

### ISSUE THREAD

**Issue by jaeyoo** [PyCQA/pylint#4684]
### Is your feature request related to a problem? Please describe

I want to have shorten form of local disable/enable pylint rules for functions.
for example, let `pylint.local_allow` is a decorator whose decorated functions disable pylint to check the specified rules at the beginning of the function code and enable the rules again. In details,

```
@pylint.local_allow(rules=["invalid-name"])  # These rules are allowed only for functions decorated by decorator `a`
def f(x):
  A = 10
  B = A*x + 20
  return B
# --> pylint ignore invalid-name for function f, but automatically re-enable the invalid-name rule.
```

This is much readable than too many:
```
# pylint:disable=invalid-name, rule2, rule3
def f(x):
  A = 10
  B = A*x + 20
  return B
# pylint:enable=invalid-name, rule2, rule3
```

when the number of functions increased, maintaining 2N lines vs N becomes quite heavy.

Also, I hope the decorator can be used for another user-defined decorator.

For example, when I have decorator `a`

```
def awesome_decorator(_func=None):
  # an awesome decorator
  f = pylint.local_allow(_func, rules=["rule1", "rule2", "rule3"])
  # add additional work for this new decorator
  return dull_decorator(f)
```

then this will reduce the codes like:

```
@pylint.local_allow(rules=["invalid-name"]) 
@dull_decorator
def f(x):
   ...
```

into

```
@awesome_decorator
def f(x):
   ...
```

### Additional context

Add any other context about the feature request here.

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for taking the time to right down a detailed issue. This seem sub-optimal in terms of performance for your program because instead of relying on comment that won't affect the runtime of your code, we'd rely on a decorator that would.

You could use ``disable`` for the function's scope instead and do:
```python
def f(x):
  # pylint:disable=invalid-name, rule2, rule3
  A = 10
  B = A*x + 20
  return B
```
